### ISSUE TITLE
The use of parenthesis; "dsl methods" and multi-line invocations

### ISSUE THREAD

**Issue by pedrorolo** [rubocop-hq/ruby-style-guide#356]
I think that the rule about dsl methods and the absense of parenthesis should be removed from this guide, since what is or not a dsl method is highly subjective.

On the other hand, I'd like to propose that multi-line method invocations should indeed use parenthesis, since it is the only way to make them be auto-indented in emacs.

**Comment by LavirtheWhiolet** [comment]
```
module MyDSL

  def task name, &action
  end
  
  def iff(condition)
  end

end
```
Or, using docs:
```
module MyDSL

  # :call-seq:
  #   task :name do ... end
  #
  def task(*args, &block)
  end
  
  # :call-seq:
  #   iff(condition).then { ... }.else { ... }
  # 
  def iff(*args, &block)
  end

end
```

**Comment by pauldraper** [comment]
> Might be a bit subjective, but it's commonly accepted.

More than a bit subjective; it's downright inconsistent.

Should it be `sleep(100)` or `sleep 100`? [This example](https://github.com/bbatsov/ruby-style-guide#rare-eol-annotations) says the latter, though IDK why.

And how are `puts` or `attr_reader` keywords? There is nothing special about them. A developer can define identically behaving `puts2` and `attr_reader2` methods.

**Comment by bbatsov** [comment]
I'm not arguing. Apart from class "macros" I use parens for pretty much everything myself. I was just saying that we can't have a community resource discourage a very common community-established practice. What we can do is to extend the rule with more information, which I think would solve the concerns that were brought up here.

**Comment by pedrorolo** [comment]
There's no such thing as class macros. There are just methods.

**Comment by bbatsov** [comment]
@pedrorolo True, but many people call them "macros". (e.g. https://codeburst.io/ruby-macros-18bb67e051c7)

**Comment by pedrorolo** [comment]
That idea, and consequently this rule, harms newcomers to the ruby language, in the sense that it passes the idea that there's something special or magic about those method invocations.
